From 1bc1091f0b9b546d7255624e0a1a79d1a0f59954 Mon Sep 17 00:00:00 2001
From: Alison Wang <alison.wang@nxp.com>
Date: Tue, 9 Oct 2018 16:43:31 +0800
Subject: [PATCH 339/706] arm64: dts: ls1028a: Update Cadence display
 controller/phy nodes

This patch updates the related nodes for Cadence display controller/phy.

Signed-off-by: Alison Wang <alison.wang@nxp.com>
(cherry picked from commit 89830c2316e64504512a95abf12b13096548fb6d)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 arch/arm64/boot/dts/freescale/fsl-ls1028a-pxp.dts | 8 ++++++++
 arch/arm64/boot/dts/freescale/fsl-ls1028a-qds.dts | 8 ++++++++
 arch/arm64/boot/dts/freescale/fsl-ls1028a-rdb.dts | 8 ++++++++
 arch/arm64/boot/dts/freescale/fsl-ls1028a.dtsi    | 8 ++++----
 4 files changed, 28 insertions(+), 4 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-ls1028a-pxp.dts b/arch/arm64/boot/dts/freescale/fsl-ls1028a-pxp.dts
index 2a710544d91a..dea813a7141a 100644
--- a/arch/arm64/boot/dts/freescale/fsl-ls1028a-pxp.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-ls1028a-pxp.dts
@@ -93,3 +93,11 @@
 		full-duplex;
 	};
 };
+
+&edp {
+	fsl,no_edid;
+	lane_mapping = <0x1b>;
+	edp_link_rate = <0x6>;
+	edp_num_lanes = <0x4>;
+	status = "okay";
+};
diff --git a/arch/arm64/boot/dts/freescale/fsl-ls1028a-qds.dts b/arch/arm64/boot/dts/freescale/fsl-ls1028a-qds.dts
index 961ba4c4ab2e..4c74664949a8 100755
--- a/arch/arm64/boot/dts/freescale/fsl-ls1028a-qds.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-ls1028a-qds.dts
@@ -240,3 +240,11 @@
 &sai1 {
 	status = "okay";
 };
+
+&edp {
+	fsl,no_edid;
+	lane_mapping = <0x1b>;
+	edp_link_rate = <0x6>;
+	edp_num_lanes = <0x4>;
+	status = "okay";
+};
diff --git a/arch/arm64/boot/dts/freescale/fsl-ls1028a-rdb.dts b/arch/arm64/boot/dts/freescale/fsl-ls1028a-rdb.dts
index 5d4fb2ef3e9f..107332e65dc3 100755
--- a/arch/arm64/boot/dts/freescale/fsl-ls1028a-rdb.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-ls1028a-rdb.dts
@@ -300,3 +300,11 @@
 	phy-handle = <&qsgmii_phy3>;
 	phy-connection-type = "qsgmii";
 };
+
+&edp {
+	fsl,no_edid;
+	lane_mapping = <0x1b>;
+	edp_link_rate = <0x6>;
+	edp_num_lanes = <0x4>;
+	status = "okay";
+};
diff --git a/arch/arm64/boot/dts/freescale/fsl-ls1028a.dtsi b/arch/arm64/boot/dts/freescale/fsl-ls1028a.dtsi
index 82e2cfc7b4dc..5b4cc3b99ada 100755
--- a/arch/arm64/boot/dts/freescale/fsl-ls1028a.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-ls1028a.dtsi
@@ -721,7 +721,7 @@
 		};
 	};
 
-	dp0: malidp@f080000 {
+	dp0: display@f080000 {
 		compatible = "arm,mali-dp500";
 		reg = <0x0 0xf080000 0x0 0x10000>;
 		interrupts = <0 222 0x4>,
@@ -734,12 +734,12 @@
 
 		port {
 			dp0_out: endpoint {
-				remote-endpoint = <&ip8501_out>;
+				remote-endpoint = <&edp_phy_out>;
 			};
 		};
 	};
 
-	ip8501: dp@f200000 {
+	edp: phy@f200000 {
 		compatible = "fsl,ls1028a-dp";
 		reg = <0x0 0xf200000 0x0 0x100000>;
 		interrupts = <0 221 0x4>;
@@ -769,7 +769,7 @@
 			      "clk_vif", "clk_apb_csr","clk_apb_ctrl";
 
 		port {
-			ip8501_out: endpoint {
+			edp_phy_out: endpoint {
 				remote-endpoint = <&dp0_out>;
 			};
 		};
-- 
2.17.1

